module-system/sov-modules-macros/src/runtime_event.rs

```diff
@@ -0,0 +1,115 @@
+use proc_macro2::Span;
+use syn::DeriveInput;
+
+use super::common::{
+    get_generics_type_param, get_serialization_attrs, StructDef, StructFieldExtractor,
+};
+
+pub(crate) const EVENT: &str = "Event";
+
+pub(crate) struct RuntimeEventMacro {
+    field_extractor: StructFieldExtractor,
+}
+
+impl<'a> StructDef<'a> {
+    fn create_event_enum_legs(&self) -> Vec<proc_macro2::TokenStream> {
+        self.fields
+            .iter()
+            .map(|field| {
+                let name = &field.ident;
+                let ty = &field.ty;
+
+                quote::quote!(
+                    #[doc = "Module event."]
+                    #name(<#ty as ::sov_modules_api::Module>::Event),
+                )
+            })
+            .collect()
+    }
+
+    fn create_get_key_string_impl(&self) -> proc_macro2::TokenStream {
+        let enum_ident = self.enum_ident(EVENT);
+
+        let match_legs: Vec<proc_macro2::TokenStream> = self
+            .fields
+            .iter()
+            .map(|field| {
+                let module_name = &field.ident;
+                let module_name_str = &field.ident.to_string();
+                quote::quote!(
+                    #enum_ident::#module_name(inner)=>{
+                        let enum_name: String = format!("{:?}", inner)
```

Comment:
Looks like we forgot to replace this with a call to `inner.event_key()`

Reply:
Done.